logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageRepository.kt

```diff
@@ -24,6 +24,7 @@ interface MessageRepository {
     suspend fun persistMessage(message: Message): Either<CoreFailure, Unit>
     suspend fun deleteMessage(messageUuid: String, conversationId: ConversationId): Either<CoreFailure, Unit>
     suspend fun deleteMessage(messageUuid: String): Either<CoreFailure, Unit>
+    suspend fun updateMessage(message:Message): Either<CoreFailure,Unit>
```

Comment:
Instead of creating a generic update function just to update the status, perhaps we could adapt the current `markMessageAsSent` and make it more flexible:

```kotlin 
suspend fun updateMessageStatus(conversationId: ConversationId, messageUuid: String, newStatus: Message.Status) 
```